python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,267 @@
+                collects the first element
+    - reverse:  gather the last `len_gather_map` rows, the first thread
+                collects the last element
+    - random:   create a pseudorandom gather map
+
+    `len_gather_map`, `len_column` gets rounded to integer.
+    """
+    len_gather_map = round(len_gather_map)
+    len_column = round(len_column)
+
+    rstate = cupy.random.RandomState(seed=0)
+    if how == "sequence":
+        return cudf.Series(cupy.arange(0, len_gather_map))
+    elif how == "reverse":
+        return cudf.Series(
+            cupy.arange(len_column - 1, len_column - len_gather_map - 1, -1)
+        )
+    elif how == "random":
+        return cudf.Series(rstate.randint(0, len_column, len_gather_map))
+
+
+def make_boolean_mask_column(size):
+    rstate = cupy.random.RandomState(seed=0)
+    return cudf.core.column.as_column(rstate.randint(0, 2, size).astype(bool))
+
+
+def accepts_cudf_fixture(
+    cls, *, dtype="int", nulls=None, cols=None, rows=None, name=None
+):
+    """Pass "standard" cudf fixtures to functions without renaming parameters.
+
+    The fixture generation logic in conftest.py provides a plethora of useful
+    fixtures to allow developers to easily select an appropriate cross-section
+    of the space of objects to apply a particular benchmark to. However, the
+    usage of this fixtures is cumbersome because creating them in a principled
+    fashion results in long names and very specific naming schemes. This
+    decorator abstracts that naming logic away from the developer, allowing
+    them to instead focus on defining the fixture semantically by describing
+    its properties.
+
+    Parameters
+    ----------
+    cls : Union[str, Type]
+        The class of object to test. May either be specified as the type
+        itself, or using the name (as a string). If a string, the case is
+        irrelevant as the string will be converted to all lowercase.
+    dtype : Union[str, Iterable[str]], default 'int'
+        The dtype or set of dtypes to use.
+    nulls : Optional[bool], default None
+        Whether to test nullable or non-nullable data. If None, both nullable
+        and non-nullable data are included.
+    cols : Optional[int], None
+        The number of columns. Only valid if cls == 'dataframe'. If None, use
+        all possible numbers of columns. Specifying multiple values is
+        unsupported.
+    rows : Optional[int], None
+        The number of rows. If None, use all possible numbers of rows.
+        Specifying multiple values is unsupported.
+    fixture : str, default None
+        The name of the fixture as used in the decorated test. If None,
+        defaults to `cls.lower()` if cls is a string, otherwise
+        `cls.__name__.lower()`. Use of this name allows the decorated function
+        to masquerade as a pytest receiving a fixture while substituting the
+        real fixture (with a much longer name).
```

Comment:
It's not immediately clear to me what this means. Can you give an example?

Reply:
Basically it means that you can write
```
@accepts_cudf_fixture(cls="column", dtype="int", nulls=False)
def test_foo(column)
```
and pytest will actually see a function
```
def test_foo(column_dtype_int_nulls_False)
``
Does that answer your question?

Reply:
I guess my question is more with why `name` argument is needed for this utility function?

Reply:
I personally find it confusing to provide a way to alias your parameter, especially when the name of `cls` is unambiguous. The only usage of this is if we have very complex union in the future that even the name of `cls` becomes non-intelligible, then we can think of introducing the aliases.

I agree with your definition of the decorator in its current state. I think the debating point is whether we should take the decorator a step further, hiding more details from the developer. It's a good addition, but probably beyond what this PR has to offer.

Reply:
Even if we have a very complex union the `cls` should never become unintelligible; that would only be possible if our class hierarchy become complex and we introduced weird classes like `IndexedFrameWithoutNulls` or something like that, which I don't think will ever happen. @shwina what do you think about removing support for the `name` parameter entirely?

Reply:
> I agree with your definition of the decorator in its current state. I think the debating point is whether we should take the decorator a step further, hiding more details from the developer (with other decorators). It's a good addition, but probably beyond what this PR has to offer.

What do you mean by this? What other details would you want to hide? Are you suggesting something like `accepts_cudf_fixture_with_nulls`, i.e. putting more of the parameter information into decorators that wrap this one? I don't think that adds any value, but perhaps I'm misunderstanding your suggestion.

Reply:
Apologies, you should read this paragraph in the context of the comments I posted in the other PR: https://github.com/rapidsai/cudf/pull/11122#discussion_r903866124, I meant specifically that the fixture that @shwina proposed is the advanced version that hides more detail.

Reply:
OK, and in that scenario it's only "hiding" in the sense of the name of the decorator, right? The function signature is not changed at all.

Reply:
Discussed offline, we are removing the name parameter but otherwise leaving the decorator as is (a single decorator with `cls` as a parameter rather than one decorator per `cls`).